src/widgets/dialogs/ColorPickerDialog.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "util/LayoutCreator.hpp"
+#include "widgets/BaseWindow.hpp"
+#include "widgets/helper/ColorButton.hpp"
+#include "widgets/helper/QColorPicker.hpp"
+
+#include <pajlada/signals/signal.hpp>
+
+namespace chatterino {
+
+/**
+ * @brief A custom color picker dialog.
+ *
+ * This class exists because QColorPickerDialog did not suit our use case.
+ * This dialog provides buttons for recently used and default colors, as well
+ * as a color picker widget identical to the one used in QColorPickerDialog.
+ */
+class ColorPickerDialog : public BaseWindow
```

Comment:
This can be made into a `BasePopup` instead of a `BaseWindow` - it will make pressing "Escape" act as pressing the "Cancel" button.

Reply:
See 61d5fdb.